lib/es_client.rb

```
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require 'typhoeus'
+require 'json'
+require 'elasticsearch'
+
+module ElasticsearchSSM
+  #
+  # Slim wrapper for common ES operations
+  #
+  class ESClient
+    attr_reader :elasticsearch_client
+
+    def initialize(es_endpoint = 'http://localhost:9200')
+      @elasticsearch_client = Elasticsearch::Client.new(url: es_endpoint, log: true)
+    end
+
+    def cluster_green?
+      elasticsearch_client.cluster.health["status"] == 'green'
+    end
+
+    def enable_shard_allocation
+      response = elasticsearch_client.cluster.put_settings(body: { transient: { 'cluster.routing.allocation.enable' => nil } })
+      raise PatchError.new("Error enabling shard allocation: #{response}", 1) if response["acknowledged"] != true
+    end
+
+    def disable_shard_allocation
+      response = elasticsearch_client.cluster.put_settings(body: { transient: { 'cluster.routing.allocation.enable' => 'none' } })
+      raise PatchError.new("Error disabling shard allocation: #{response}", 1) if response["acknowledged"] != true
+    end
+
+    def stop_elasticsearch
+      ShellClient.run_shell_command('sudo monit stop elasticsearch')
```

Is there anyway to indicate if this succeeded or failed? What would happen if this failed silently? Would we end up installing patches but not restarting the host? Is this ok?

---

Monit itself does not return any output on success (if you ran this command in shell yourself you wouldn't get anything. I will add -v, the stdout is logged.

For failures, ShellClient raises a `PatchError` and corresponding outputs and status codes are logged. So this cannot fail silently.